Userland/Applications/PixelPaint/ImageEditor.cpp

```diff
@@ -227,6 +231,17 @@ void ImageEditor::mousemove_event(GUI::MouseEvent& event)
 
 void ImageEditor::mouseup_event(GUI::MouseEvent& event)
 {
+    if (m_middle_mouse_down && event.button() == GUI::MouseButton::Middle)
+        m_middle_mouse_down = false;
+
+    if (m_space_down && event.button() == GUI::MouseButton::Left) {
+        m_active_cursor = Gfx::StandardCursor::Hand;
+    } else if (event.button() == GUI::MouseButton::Middle) {
+        if (m_space_down)
+            m_active_cursor = event.buttons() ? Gfx::StandardCursor::Drag : Gfx::StandardCursor::Hand;
```

Comment:
This logic seems very convoluted, `event.buttons()` should always be true if `event.button() == GUI::MouseButton::Middle`